Client/src/StoreModules/QuestionStoreModule.ts

```diff
@@ -617,22 +617,36 @@ async function loadQuestion(
   initialParamData?: Record<string, string>,
 ) {
   const step = stepId ? await wdkService.findStep(stepId) : undefined;
+  const initialParams = step ? initialParamDataFromStep(step) : initialParamData != null ? initialParamData : {};
+
   try {
-    const question = step == null
-      ? await wdkService.getQuestionAndParameters(searchName)
-      : await wdkService.getQuestionGivenParameters(searchName, step.searchConfig.parameters);
+    const question = Object.keys(initialParams).length > 0
+      ? await wdkService.getQuestionGivenParameters(searchName, initialParams)
+      : await wdkService.getQuestionAndParameters(searchName);
+
     const recordClass = await wdkService.findRecordClass(question.outputRecordClassName);
-    const paramValues = step != null ? integrateStepAndDefaultParamValues(question.parameters, step)
-                      : initialParamData != null ? extractRealParamValues(question.parameters, initialParamData)
-                      : makeDefaultParamValues(question.parameters)
+    const paramValues = question.parameters.reduce(function(values, { name, initialDisplayValue, type }) {
+      return Object.assign(
+        values,{
+          [name]: (
+            (step == null && type === 'input-step')
+            ? ''
+            : (name in initialParams)
+              ? initialParams[name]
+              : initialDisplayValue
+          )
+        }
+      );
+    }, {} as ParameterValues);
+
```

Comment:
This was previously factored into a separate function. Could you make it a separate function again?